File: net/NetTupleEngine/Atom.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace it.mintlab.desktopnet.tupleengine
{
    public class Atom : Term, IRefObject
    {
        private string value;
        private object refObject;
        private bool withRefObject;

        public Atom(string value)
        {
            this.value = value;
            withRefObject = false;
        }

        public void setRefObject(object obj)
        {
            refObject = obj;
            withRefObject = true;
        }

        public override string ToString()
        {
            return value;
        }

        #region Term Membri di

        internal override bool unify(Term t, List<Var> vars1, List<Var> vars2)
        {
            if (t is Atom)
            {
                Atom ta = (Atom)t;
                if (this.value.Equals(ta.value))
                {
                    return true;
                }
            }
            if (t is Var)
            {
                return t.unify(this, vars2, vars1);
            }
            return false;
        }

        public override object getRefObject()
        {
            if(withRefObject)
                return refObject;
            else
                if(value.StartsWith("'"))
                    return value.Replace("'","");
                else if(value.StartsWith("\""))
                    return value.Replace("\"","");
                else
                    return value;
        }

        #endregion

    }
}

File: trunk/java/JavaTupleEngine/src/it/mintlab/desktopjava/tupleengine/Atom.java
package it.mintlab.desktopjava.tupleengine;

	public class Atom extends Term implements IRefObject
	{
		private String value;
		private Object refObject;
		private boolean withRefObject;

		public Atom(String value)
		{
			this.value = value;
			withRefObject = false;
		}

		public final void setRefObject(Object obj)
		{
			refObject = obj;
			withRefObject = true;
		}

		@Override
		public String toString()
		{
			return value;
		}

		@Override
		public boolean unify(Term t, java.util.ArrayList<Var> vars1, java.util.ArrayList<Var> vars2)
		{
			if (t instanceof Atom)
			{
				Atom ta = (Atom)t;
				if (this.value.equals(ta.value))
				{
					return true;
				}
			}
			if (t instanceof Var)
			{
				return t.unify(this, vars2, vars1);
			}
			return false;
		}

		@Override
		public Object getRefObject()
		{
			return withRefObject ? refObject : value;
		}

	}